### PR TITLE
Add parental rating fields to EPG and Recordings (Piers).

### PR DESCRIPTION
**Pull request by DeltaMikeCharlie**
Piers replacement for https://github.com/kodi-pvr/pvr.demo/pull/137.
v22.2.0

Add parental rating fields in support of https://github.com/xbmc/xbmc/pull/24096

Access via $INFO[ListItem.ParentalRating], $INFO[ListItem.ParentalRatingCode], $INFO[ListItem.ParentalRatingIcon] and $INFO[ListItem.ParentalRatingSource] respectively.

Note: 'ListItem.ParentalRatingCode' will be available once https://github.com/xbmc/xbmc/pull/23586 has been merged. However, the columns are already in the epgtags table of the PVR/EPG database.

### CODE REVIEW

File: pvr.demo/changelog.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+v22.2.0
+- Add support for parentalRating+Code+Icon+Source to EPG, timers and recordings.
```
Comment (ksooo) [inline pvr.demo/changelog.txt]: Remove "timers", as this is not true.

File: pvr.demo/PVRDemoAddonSettings.xml [hunk 2]
```diff
@@ -1194,5 +1273,18 @@
...
+        <parentalrating>13</parentalrating>
+        <parentalratingcode>CC-DD</parentalratingcode>
+        <parentalratingicon>data/cc-dd.png</parentalratingicon>
+        <parentalratingsource>ABC</parentalratingsource>
+    </timer>
```
Comment (ksooo) [inline pvr.demo/PVRDemoAddonSettings.xml]: Remove this. We have no parental rating support for timers.

File: src/PVRDemo.h [hunk 3]
```diff
@@ -77,6 +84,10 @@ struct PVRDemoTimer
   PVR_TIMER_STATE state;
   std::string strTitle;
   std::string strSummary;
+  int iParentalRating = 0;
+  std::string strParentalRatingCode;
+  std::string strParentalRatingIcon;
+  std::string strParentalRatingSource;
```
Comment (ksooo) [inline src/PVRDemo.h]: Remove this!

File: src/PVRDemo.cpp [hunk 4]
```diff
@@ -916,6 +954,21 @@ bool CPVRDemo::ScanXMLTimerData(const XMLNode* pTimerNode, PVRDemoTimer& timer)
...
+    timer.strParentalRatingCode = strTmp;
+
+  /* parental rating icon */
+  if (XMLGetString(pTimerNode, "parentalratingicon", strTmp))
+    timer.strParentalRatingIcon = ClientPath() + strTmp;
+
+  /* parental rating source */
+  if (XMLGetString(pTimerNode, "parentalratingsource", strTmp))
+    timer.strParentalRatingSource = strTmp;
+
```
Comment (ksooo) [inline src/PVRDemo.cpp]: Remove this.

**Review by ksooo — CHANGES_REQUESTED** [review]
Please remove the timer leftovers. Add-On API does not offer parental rating data fields for timers. This makes no sense as we found out in earlier discussions.

File: pvr.demo/changelog.txt (hunk 1 above)
Comment (DeltaMikeCharlie) [reply pvr.demo/changelog.txt]: Done.

File: pvr.demo/PVRDemoAddonSettings.xml (hunk 2 above)
Comment (DeltaMikeCharlie) [reply pvr.demo/PVRDemoAddonSettings.xml]: Done.

File: src/PVRDemo.h (hunk 3 above)
Comment (DeltaMikeCharlie) [reply src/PVRDemo.h]: Done.

File: src/PVRDemo.cpp (hunk 4 above)
Comment (DeltaMikeCharlie) [reply src/PVRDemo.cpp]: Done.

**Review by ksooo — APPROVED** [review]

**Pull request merged**